aspnet/fundamentals/dependency-injection.rst

```diff
@@ -1,10 +1,291 @@
+
+First, add the appropriate container to the dependencies property in project.json:
+
+.. code-block:: javascript
+
+	"dependencies" : {
+		"StructureMap.Dnx": "0.4.0-alpha"
+	},
+
+.. note:: Currently StructureMap does not support .NET Core.
+
+Next, configure the container in ``ConfigureServices`` and return an ``IServiceProvider``:
+
+.. code-block:: c#
+
+	public IServiceProvider ConfigureServices(IServiceCollection services)
+	{
+		services.AddMvc();
+		// add other services
+		
+		// Add StructureMap
+		var container = new StructureMap.Container();
+		container.Populate(services);
+		return container.GetInstance<IServiceProvider>();
+	}
+
+.. note:: If you add StructureMap to the sample project, you will need to add a ``[DefaultConstructor]`` attribute to the default constructor for ``Operation``.
+
+.. list-table:: ASP.NET 5 / DNX Containers (in alphabetical order)
+   :header-rows: 1
+
+   * - Package (Nuget)
+     - Setup Instructions
+   * - `Autofac.Dnx <http://www.nuget.org/packages/Autofac.Dnx/>`_
+     - `Autofac.org <http://autofac.org/>`_
+   * - `StructureMap.Dnx <http://www.nuget.org/packages/StructureMap.Dnx>`_
+     - `Getting Started <https://github.com/structuremap/structuremap.dnx/blob/master/README.md>`_
+
+Recommendations
+---------------
+
+When working with dependency injection, keep the following recommendations in mind:
+
+- DI is for objects that have complex dependencies. Controllers, services, adapters, and repositories are all examples of objects that might be added to DI.
+- Avoid storing data and configuration directly in DI. For example, a user's shopping cart shouldn't typically be added to the services container. Configuration should use the :doc:`options`. Similarly, avoid "data holder" objects that only exist to allow access to some other object. It's better to request the actual item needed via DI, if possible.
+- Avoid static access to services.
+- Avoid service location in your application code.
+- Avoid static access to ``HttpContext``.
+
+.. note:: There are exceptions to these recommendations, for instance MVC internally uses the ``IActionContextAccessor`` service to access very short-lived ``ActionContext`` instances. In typical applications, the need for this kind of service should be rare.
+
+Remember, dependency injection is an *alternative* to static/global object access patterns. You will not be able to realize the benefits of DI if you mix it with static object access.
```

Comment:
Well, it applies to the last 3 bullets, which is why I made it a separate, overall recommendation.